core/example/longrange/ewald.cpp

```diff
@@ -103,210 +76,428 @@ double TEwald::compute( ParticleList &particles, double lx, double ly,
+
+        std::cout << "real-space contribution: "
+                  << ( ns_elapsed_r.count() / 1000000000.0 ) << " s " << Ur
+                  << std::endl;
+    }
+    else
+    {
+
+        // TODO: cellsize dynamic!
+        double lc = 1.0;
+        double cfactor = std::ceil( r_max / lc );
+        int nneigdim = 2 * (int)cfactor + 1;
+        int nneig = nneigdim * nneigdim * nneigdim;
+
+        std::cout << "l: " << lx << " cfactor: " << cfactor << " lc: " << lc
+                  << " r_max: " << r_max << std::endl;
+
+        int n_cells =
+            std::ceil( lx / lc ) * std::ceil( ly / lc ) * std::ceil( lz / lc );
+        Kokkos::View<int *, MemorySpace> cdim( "cdim", 3 );
+        cdim( 0 ) = std::ceil( lx / lc );
+        cdim( 1 ) = std::ceil( ly / lc );
+        cdim( 2 ) = std::ceil( lz / lc );
+
+        auto start_time_rsort = std::chrono::high_resolution_clock::now();
+        // create a Cabana linked list
+        // lower end of system
+        double grid_min[3] = {0.0, 0.0, 0.0};
+        // upper end of system
+        double grid_max[3] = {lx, ly, lz};
+        // cell size
+        double grid_delta[3] = {lc, lc, lc};
+        // create LinkedCellList
+        Cabana::LinkedCellList<MemorySpace> cell_list( r, grid_delta, grid_min,
+                                                       grid_max );
+
+        // resort particles in original ParticleList to be sorted according to
+        // LinkedCellList
+        Cabana::permute( cell_list, particles );
+        auto end_time_rsort = std::chrono::high_resolution_clock::now();
+        auto elapsed_time_rsort = end_time_rsort - start_time_rsort;
+
+        auto start_time_rij = std::chrono::high_resolution_clock::now();
+
+        // energy contribution from neighbor cells
+        double Ur_ij = 0.0;
+
+        for ( int n = nneig / 2 + 1; n < nneig; ++n )
+        {
+            double Ur_ijn;
+            Kokkos::parallel_reduce(
+                n_cells,
+                KOKKOS_LAMBDA( const int idx, double &Ur_ic ) {
+                    // get coordinates of local domain
+                    int ix, iy, iz;
+                    cell_list.ijkBinIndex( idx, ix, iy, iz );
+
+                    // index of neighbor (to dermine which neighbor it is
+                    int jcidx = n;
+
+                    // compute position in relation to local cell
+                    int jnx = jcidx % nneigdim - (int)cfactor;
+                    int jny = ( jcidx / nneigdim ) % nneigdim - (int)cfactor;
+                    int jnz = jcidx / ( nneigdim * nneigdim ) - (int)cfactor;
+
+                    // compute global cell index
+                    int jc = ( iz + jnz + cdim( 2 ) ) % cdim( 2 ) +
+                             ( iy + jny + cdim( 1 ) ) % cdim( 1 ) * cdim( 2 ) +
+                             ( ix + jnx + cdim( 0 ) ) % cdim( 0 ) * cdim( 1 ) *
+                                 cdim( 2 );
+
+                    // get neighbor coodinates
```

Comment:
Nice descriptive comments here. I'd like to see a bit more later in this loop

Reply:
Updated the comments for the loops for the interactions between surrounding boxes and local boxes.